nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+          description = ''
+            The webserver used for the Sympa web interface. Set it to `none` if you want to configure it yourself.
+            Further nginx configuration can be done by adapting <literal>services.nginx.virtualHosts.&lt;name&gt;</literal>.
+          '';
+        };
+
+        https = mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            Whether to use HTTPS. When nginx integration is enabled, this option forces SSL and enables ACME.
+            Please note that Sympa web interface always uses https links even when this option is disabled.
+          '';
+        };
+
+        fcgiProcs = mkOption {
+          type = types.ints.positive;
+          default = 2;
+          description = "Number of FastCGI processes to fork.";
+        };
+      };
+
+      mta = {
+        type = mkOption {
+          type = types.enum [ "postfix" "none" ];
+          default = "postfix";
+          description = ''
+            Mail transfer agent (MTA) integration. Use `none` if you want to configure it yourself.
+
+            The `postfix` integration sets up local Postfix instance that will pass incoming
+            messages from configured domains to Sympa. You still need to configure at least
+            outgoing message handling using e.g. <option>services.postfix.relayHost</option>.
+          '';
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "
+          Extra lines to be added verbatim to the main configuration file.
+        ";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+
+      environment = {
+        systemPackages = [ pkg ];
+      };
+
+      users.users = optional (user == "sympa")
+        { name = "sympa";
+          description = "Sympa mailing list manager user";
+          uid = config.ids.uids.sympa;
+          group = group;
+          home = cfg.dataDir;
+          createHome = true;
+        };
+
+      users.groups =
+        optional (group == "sympa")
+        { name = group;
+          gid = config.ids.gids.sympa;
+        };
+
+      warnings = optional (cfg.database.password != "")
+        ''config.services.sympa.database.password will be stored as plaintext
+          in the Nix store. Use database.passwordFile instead.'';
+
+      # Create database passwordFile default when password is configured.
+      services.sympa.database.passwordFile =
+        (mkDefault (toString (pkgs.writeTextFile {
+          name = "sympa-database-password";
+          text = cfg.database.password;
+        })));
+
+      systemd.tmpfiles.rules = [
+        "d '/run/sympa' 0755 ${user} ${group} - -"
+      ];
+
+      systemd.services.sympa = {
+        description = "Sympa mailing list manager";
+
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        path = [ pkg ];
```

Comment:
Why put the package into $PATH and then later using `${pkg}` in the script? You should drop one of them